src/processTargets/processSelectionType.ts

```diff
@@ -173,6 +175,78 @@ function processParagraph(
   };
 }
 
+function processNonWhitespaceSequence(
+  target: PrimitiveTarget,
+  selection: SelectionWithEditor,
+  selectionContext: SelectionContext
+) {
+  const { selectionType, insideOutsideType, position, modifier } = target;
+  const { document } = selection.editor;
+  const { start, end } = selection.selection;
+  let newSelection;
+
+  let startLine;
+  let i = start.line;
+  do {
+    startLine = document.lineAt(i++);
+  } while (startLine.isEmptyOrWhitespace && i <= end.line);
+
+  // Can't find any non whitespace characters in the selection
+  if (startLine.isEmptyOrWhitespace) {
```

Comment:
This implementation seems overly complicated. What's wrong with the approach proposed in the issue description?

Reply:
I would say it follows the issue exactly. . The implementation needs to be this complex to get that functionality